### DISCUSSION TITLE
Usable w/ Terminal.Gui v2?

### DISCUSSION THREAD

**Discussion by wobbet** [tznind/TerminalGuiDesigner#293] [General]
I'm starting work on a Terminal.Gui v2 project and would like to use this designer with it. Is there a version (the v2 branch maybe?) that supports the API changes in the Terminal.Gui v2 library?

Thx!

**Comment by tznind** [comment]
### The long answer
Certainly! the [v2](https://github.com/gui-cs/TerminalGuiDesigner/tree/v2) branch is stable.  You should clone the repo and run locally.  There is no dotnet tool release since v2 is still pre-alpha.

To use the v2 branch of the designer you should ideally reference the same nuget package as the designer's csproj - see [TerminalGuiDesigner.csproj](https://github.com/gui-cs/TerminalGuiDesigner/blob/v2/src/TerminalGuiDesigner.csproj)

At time of writing this is:

```
<PackageReference Include="Terminal.Gui" Version="2.0.0-pre.1491" />
```

### The short answer
Yes, use v2 branch but make sure to reference the same nuget package as designer.

### Updates
I try to keep v2 branch up-to-date but it is always slightly behind the latest Terminal.Gui alpha nuget release.  This is because I need to handle renamed variables, new types and do QA etc.

The current v2 update PR is #292 which will bring the nuget package compatibility up to `2.0.0-v2-develop.2168` (at time of writing).  This update branch is unstable and I don't advise you use it.

Each PR into `v2` branch is thoroughly checked to ensure it is compatible with the library nuget package at the point of merge.  

After merging #292  into `v2` - when you pull the new changes, you may need to make small changes to your `.Designer.cs` files.  You would need these changes anyway to reference the latest nuget package.

TerminalGuiDesigner uses the tool for its own UIs (it writes itself!) so you can see examples of the .Designer.cs file changes needed, they are pretty minor e.g.

![image](https://github.com/user-attachments/assets/0d1c5b78-a876-4423-9e13-77cdf1fb3371)
_One of the v2 changes that is not backwards compatible was `ListView` changing to use `ObservableList` and generic `ListWrapper<T>`.  This is the .Designer.cs change you need to make to update your old source.  New source generated with the tool will be correct_

**Reply by tznind** [reply]
@wobbet I have released v2 as a pre-release to nuget.org.  You can install it with:

For convenience the home screen now shows the corresponding compatible nuget version you should reference:

```
dotnet tool install --global TerminalGuiDesigner --prerelease
```

![image](https://github.com/user-attachments/assets/efb8552b-8d12-46f7-a5d6-4a0020ac9c8a)
